`statusoptional` linter panics

I tried bumping KAL in today in ORC and still got a panic in `statusoptional`, even with https://github.com/kubernetes-sigs/kube-api-linter/pull/84:
```
❯ make lint
/home/martin/dev/openshift/openstack-resource-controller/bin/golangci-lint custom
/home/martin/dev/openshift/openstack-resource-controller/bin/golangci-kube-api-linter run
ERRO [runner] Panic: statusoptional: package "v1alpha1" (isInitialPkg: true, needAnalyzeSource: true): runtime error: invalid memory address or nil pointer dereference: goroutine 21678 [running]:
runtime/debug.Stack()
        /usr/lib/golang/src/runtime/debug/stack.go:26 +0x5e
github.com/golangci/golangci-lint/v2/pkg/goanalysis.(*action).analyzeSafe.func1()
        /tmp/custom-gcl2564857524/golangci-lint/pkg/goanalysis/runner_action.go:50 +0x257
panic({0x1ba11c0?, 0x2b33530?})
        /usr/lib/golang/src/runtime/panic.go:792 +0x132
sigs.k8s.io/kube-api-linter/pkg/analysis/statusoptional.(*analyzer).checkStatusStruct(0x0?, 0x0?, 0x10000?, {0x20b9900?, 0xc034d3a0a8?}, {0x20acce0?, 0xc0210ffdc8?})
        /home/martin/go/pkg/mod/sigs.k8s.io/kube-api-linter@v0.0.0-20250516124914-6df69a12c92c/pkg/analysis/statusoptional/analyzer.go:99 +0x3a
sigs.k8s.io/kube-api-linter/pkg/analysis/statusoptional.(*analyzer).checkStatusStruct(0xc000d3aa00, 0xc01605f7a0, 0xa8c882?, {0x20b9900, 0xc034d3a0a8}, {0x20acce0, 0xc0210ffdc8})
        /home/martin/go/pkg/mod/sigs.k8s.io/kube-api-linter@v0.0.0-20250516124914-6df69a12c92c/pkg/analysis/statusoptional/analyzer.go:117 +0x28b
sigs.k8s.io/kube-api-linter/pkg/analysis/statusoptional.(*analyzer).run.func1(0xb0398d?, {0x1?, 0x200000?, 0x8?}, {{0xc0273fb647, 0x6}, 0x0, 0x1, 0x0, 0x0, ...}, ...)
        /home/martin/go/pkg/mod/sigs.k8s.io/kube-api-linter@v0.0.0-20250516124914-6df69a12c92c/pkg/analysis/statusoptional/analyzer.go:92 +0xb7
sigs.k8s.io/kube-api-linter/pkg/analysis/helpers/inspector.(*inspector).InspectFields.func1({0x20b17d8, 0xc02caf3340}, 0x94?, {0xc00af55e80, 0x6, 0x8})
        /home/martin/go/pkg/mod/sigs.k8s.io/kube-api-linter@v0.0.0-20250516124914-6df69a12c92c/pkg/analysis/helpers/inspector/inspector.go:106 +0x1d1
golang.org/x/tools/go/ast/inspector.(*Inspector).WithStack(0xc00af602d0, {0xc0049b0b30?, 0x1cbe0e0?, 0xa9923f?}, 0xc03b0f1b40)
        /home/martin/go/pkg/mod/golang.org/x/tools@v0.32.0/go/ast/inspector/inspector.go:181 +0x18d
sigs.k8s.io/kube-api-linter/pkg/analysis/helpers/inspector.(*inspector).InspectFields(0x2b3d4a0?, 0x1bc19c0?)
        /home/martin/go/pkg/mod/sigs.k8s.io/kube-api-linter@v0.0.0-20250516124914-6df69a12c92c/pkg/analysis/helpers/inspector/inspector.go:61 +0x53
sigs.k8s.io/kube-api-linter/pkg/analysis/statusoptional.(*analyzer).run(0xc000d3aa00, 0xc01605f7a0)
        /home/martin/go/pkg/mod/sigs.k8s.io/kube-api-linter@v0.0.0-20250516124914-6df69a12c92c/pkg/analysis/statusoptional/analyzer.go:82 +0x147
github.com/golangci/golangci-lint/v2/pkg/goanalysis.(*action).analyze.func3(...)
        /tmp/custom-gcl2564857524/golangci-lint/pkg/goanalysis/runner_checker.go:182
github.com/golangci/golangci-lint/v2/pkg/goanalysis.(*action).analyze(0xc0044e3c20)
        /tmp/custom-gcl2564857524/golangci-lint/pkg/goanalysis/runner_checker.go:208 +0xb62
github.com/golangci/golangci-lint/v2/pkg/timeutils.(*Stopwatch).TrackStage(0xc001a04370, {0x1e01763, 0xe}, 0xc00265af48)
        /tmp/custom-gcl2564857524/golangci-lint/pkg/timeutils/stopwatch.go:111 +0x44
github.com/golangci/golangci-lint/v2/pkg/goanalysis.(*action).analyzeSafe(0x0?)
        /tmp/custom-gcl2564857524/golangci-lint/pkg/goanalysis/runner_action.go:54 +0x71
github.com/golangci/golangci-lint/v2/pkg/goanalysis.(*loadingPackage).analyze.func2(0xc0044e3c20)
        /tmp/custom-gcl2564857524/golangci-lint/pkg/goanalysis/runner_loadingpackage.go:83 +0xa5
created by github.com/golangci/golangci-lint/v2/pkg/goanalysis.(*loadingPackage).analyze in goroutine 1045
        /tmp/custom-gcl2564857524/golangci-lint/pkg/goanalysis/runner_loadingpackage.go:78 +0x1e9 
WARN [runner] Can't run linter goanalysis_metalinter: goanalysis_metalinter: statusoptional: package "v1alpha1" (isInitialPkg: true, needAnalyzeSource: true): runtime error: invalid memory address or nil pointer dereference 
ERRO Running error: can't run linter goanalysis_metalinter
goanalysis_metalinter: statusoptional: package "v1alpha1" (isInitialPkg: true, needAnalyzeSource: true): runtime error: invalid memory address or nil pointer dereference 
make: *** [Makefile:134: lint] Error 3
```

--- a/pkg/analysis/statusoptional/analyzer.go
+++ b/pkg/analysis/statusoptional/analyzer.go
@@ -85,10 +85,6 @@ func (a *analyzer) run(pass *analysis.Pass) (any, error) {
 		}
 
 		statusStructType := getStructFromField(pass, field)
-		if statusStructType == nil {
-			return
-		}
-
 		a.checkStatusStruct(pass, statusStructType, markersAccess, jsonTags)
 	})
 
@@ -96,7 +92,7 @@ func (a *analyzer) run(pass *analysis.Pass) (any, error) {
 }
 
 func (a *analyzer) checkStatusStruct(pass *analysis.Pass, statusType *ast.StructType, markersAccess markershelper.Markers, jsonTags extractjsontags.StructFieldTags) {
-	if statusType.Fields == nil || statusType.Fields.List == nil {
+	if statusType == nil || statusType.Fields == nil || statusType.Fields.List == nil {
 		return
 	}
 

